.github/workflows/java-windows.yml

```diff
@@ -17,12 +19,38 @@ jobs:
                 with:
                     java-version: ${{ matrix.java_version }}
 
+            -   name: Cache gradle dependencies
+                uses: actions/cache@v1
+                with:
+                    path: ~/.gradle/caches
+                    key: ${{ runner.os }}-gradle-${{matrix.java_version}}-${{ hashFiles('**/*.gradle*') }}
+                    restore-keys: |
+                        ${{ runner.os }}-gradle-
+
+            -   name: Cache maven dependencies
+                uses: actions/cache@v1
+                with:
+                    path: ~/.m2
+                    key: ${{ runner.os }}-maven-${{matrix.java_version}}-${{ hashFiles('**/pom.xml') }}
+                    restore-keys: |
+                        ${{ runner.os }}-maven-
+
+            -   name: Cache android jar
+                id: android-cache
+                uses: actions/cache@v1
+                with:
+                    path: android-jar-cache
+                    key: ${{ runner.os }}-android-version-${{matrix.java_version}}-${{ env.android-version }}
+
             -   name: buildNonDeprecatedJvm
                 run: ./gradlew buildNonDeprecatedJvm
 
-            -   name: download Android SDK's android.jar
+            -   name: download Android SDK's android.jar (or use cached file)
+                if: steps.android-cache.outputs.cache-hit != 'true'
```

Comment:
it looks like it always downloads and never uses the cache. But maybe due to ${{matrix.java_version}}, see if this already helps

Reply:
it looks working here: https://github.com/robstoll/atrium/pull/500/checks?check_run_id=688503736

Reply:
Nice, let's see if it still does in the latest build

Reply:
It still does :+1: